src/platform/Zephyr/BLEAdvertisingArbiter.cpp

```diff
@@ -108,6 +107,13 @@ CHIP_ERROR InsertRequest(Request & request)
 void CancelRequest(Request & request)
 {
     const bool isTopPriority = (sys_slist_peek_head(&sRequests) == &request);
+
+    // If cancelled request was top-priority, stop the advertising.
+    if (isTopPriority)
+    {
+        NotifyAdvertisingStopped(sys_slist_peek_head(&sRequests));
```

Comment:
I don't know whether this is right or not, (what's the API contract about notifying the consumer about advertising stop on a thing the consumer is canceling anyway?) but this does not address the question of what happend when InsertRequest inserts a new thing at the head....